### ISSUE TITLE
Error UnpicklingError: invalid load key, '<'. at the start of [training]

### ISSUE THREAD

**Issue by Rleuc** [BenAAndrew/Voice-Cloning-App#107]
Hello - having issues with the training step of the process on the google colab notebook.  I've tried 2 different datasets both created with the latest windows version.  All prior steps in the colab execute good.  Dataset copied to google drive as a zip and unzipped files with no difference. Runtime set as GPU.
Any help is appreciated.  Thx.

INFO:root:Setting batch size to 38, learning rate to 0.0003082207001484488. (15GB GPU memory free)
INFO:root:Loading model...
INFO:root:Loaded model
INFO:root:Loading data...
2 train files, 1 test files
INFO:root:Loaded data
---------------------------------------------------------------------------
UnpicklingError                           Traceback (most recent call last)
[<ipython-input-4-9bb687479d52>](https://localhost:8080/#) in <module>()
     20     iters_per_checkpoint=checkpoint_frequency,
     21     iters_per_backup_checkpoint=backup_checkpoint_frequency,
---> 22     train_size=1-validation_size,
     23 )

3 frames
[/usr/local/lib/python3.7/dist-packages/torch/serialization.py](https://localhost:8080/#) in _legacy_load(f, map_location, pickle_module, **pickle_load_args)
    775             "functionality.")
    776 
--> 777     magic_number = pickle_module.load(f, **pickle_load_args)
    778     if magic_number != MAGIC_NUMBER:
    779         raise RuntimeError("Invalid magic number; corrupt file?")

UnpicklingError: invalid load key, '<'.

**Comment by mellow-tnk** [comment]
oh good, someone else is getting this error too. I'm getting the same error as well.

INFO:root:Setting batch size to 28, learning rate to 0.0002645751311064591. (15GB GPU memory free)
INFO:root:Loading model...
INFO:root:Loaded model
INFO:root:Loading data...

1690 train files, 423 test files

INFO:root:Loaded data

---------------------------------------------------------------------------

UnpicklingError                           Traceback (most recent call last)

[<ipython-input-16-9bb687479d52>](https://localhost:8080/#) in <module>()
     20     iters_per_checkpoint=checkpoint_frequency,
     21     iters_per_backup_checkpoint=backup_checkpoint_frequency,
---> 22     train_size=1-validation_size,
     23 )

3 frames

[/usr/local/lib/python3.7/dist-packages/torch/serialization.py](https://localhost:8080/#) in _legacy_load(f, map_location, pickle_module, **pickle_load_args)
    775             "functionality.")
    776 
--> 777     magic_number = pickle_module.load(f, **pickle_load_args)
    778     if magic_number != MAGIC_NUMBER:
    779         raise RuntimeError("Invalid magic number; corrupt file?")

UnpicklingError: invalid load key, '<'.

**Comment by BenAAndrew** [comment]
Hi @Rleuc and @mellow-tnk. Please could you link me to a zip of your dataset for me to test with (or send to me via email if you'd like to keep it private)

**Comment by Rleuc** [comment]
@BenAAndrew 
I sent you a link to goole drive file with my  dataset.

@mellow-tnk 
Would like to try your soln, but how exactly do you run a collab offline?

Thanks!

**Comment by mellow-tnk** [comment]
@Rleuc 

just install it to your computer using the installation guide. There's a window executable you can use.
https://github.com/BenAAndrew/Voice-Cloning-App/blob/main/install.md

**Comment by Rleuc** [comment]
Thanks @mellow-tnk 
Unfortunately working with AMD graphics card and local training not an option with CPU only version.  

@BenAAndrew I tried with the full David A. dataset and get the same error.

**Comment by Rleuc** [comment]
@BenAAndrew Hi, would it be possible to create an empty or initial checkpoint file manually to get past this error? If so, what is the format? I just don't have a local nvidia card to start. 
Thanks

**Comment by mellow-tnk** [comment]
uberduck is supporting this. you can go to their discord and use their other voice cloning colab in the meantime. It's in the #notebook room and there's a lot of active participation and people helping each other.

**Comment by steve4655** [comment]
> @Rleuc uberduck is supporting this. you can go to their discord and use their other voice cloning colab in the meantime. It's in the #notebook room and there's a lot of active participation and people helping each other.

can u send the link

**Comment by AdityaThakur1** [comment]
@steve4655 @Rleuc 
I have found a solution to this. Take the pretrained.pt file that gets downloaded in the notebook files section and place into the checkpoints folder in your drive. Also, the transfer learning path should be set to 'None'. I also had to edit the checkpoint path in the notebook as the form was not giving an option to select from the drop-down. 
This should get the training started.

**Comment by g2zer0** [comment]
just using 'transfer_learning_path=None' did the job for me!
thanks :)

**Comment by colossatr0n** [comment]
@g2zer0 **Using transfer_learning_path is very beneficial to improving the quality of your model.** Setting the transfer learning path to `None` will work, but it's not recommended. I've been training a model for quite some time now without the transfer learning and though the sound of voice sounds great, all it says is gibberish.

The cause of the pickling error is because the automatic downloading that happens in the Colab with gdown isn't working properly. The pretrained model is too large for virus scanning, so instead of downloading the model, gdown downloads [Google Drive's HTML virus scanning message](https://github.com/Wanger-SJTU/FCN-in-the-wild/blob/master/FCN/pretrain_model/vgg16_from_caffe.pth). 

You can verify this by going into the Colab file explorer and downloading the failed model from `/content/Voice-Cloning-App/pretrained.pt`. It'll be there after running the "Install and import ML packages" section.

The gdown maintainers are aware that large files can't be downloaded with gdown, and you can read about it [here](https://github.com/wkentaro/gdown/issues/26).

In short, download pretrained.pt manually in a browser, put it in your drive somewhere, and change the `transfer_learning_path` to that new location.

**Comment by g2zer0** [comment]
@colossatr0n ah, that explains the gibberish result :) thank you for explaining.
